pkg/field/apply.go

```diff
@@ -176,11 +260,50 @@ func parseJSONPath(jsonPath string) (ytbx.Path, error) {
 	return result, nil
 }
 
-func parseJSONPathNamed(sc *scanner.Scanner) (ytbx.PathElement, error) {
-	if sc.Scan() != '@' {
-		return ytbx.PathElement{}, fmt.Errorf("invalid jsonpath")
+func parseJSONPathBracket(sc *scanner.Scanner) (ytbx.PathElement, error) {
+	sc.Scan()
+	t := sc.TokenText()
+	if t == "@" {
+		return parseJSONPathNamed(sc)
 	}
 
+	match, err := regexp.MatchString(`^\d+$`, t)
```

Comment:
Define the regex at the package level with `regex.MustCompile`. Then we don't recompile it at every invocation.